### PR TITLE
chore: Converts resource blocks to resource attributes in data source schema auto-generation

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Converts resource blocks to resource attributes in data source schema auto-generation. In this way singular data source can be nested in `Results` attribute in plural data source.

Link to any related issue(s): CLOUDP-290561

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals I have added appropriate changelog entries.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/common/conversion/schema_generation_test.go [hunk 1]
```diff
@@ -265,6 +240,191 @@ func TestDataSourceSchemaFromResource(t *testing.T) {
 	assert.Equal(t, expected, ds)
 }
 
+func TestDataSourceSchemaFromResource_blocksToAttrs(t *testing.T) {
```
Comment (EspenAlbert) [inline internal/common/conversion/schema_generation_test.go:243]: Nice. Looks like we are covering most cases here.
[nit] A bit long names 😅 `block block` could be replaced with bb?

**Review by EspenAlbert — APPROVED** [review]
Good job! Will this work out of the box with validators/plan_modifiers, etc.?

File: internal/common/conversion/schema_generation_test.go (hunk 1 above)
Comment (lantoli) [reply internal/common/conversion/schema_generation_test.go:243]: it's true that they could be shorter, but these long names really helped while developing the functionality. 

changed in: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2904/commits/99a6b0db0aabb6cc971d9271faac5d99a5745586

**Comment by lantoli** [comment]
> Good job! Will this work out of the box with validators/plan_modifiers, etc.?
@EspenAlbert yes, it works in the sense that they're ignored and not passed to datasource schema

**Pull request merged**